metricbeat/module/system/process/network/tracker_types.go

```diff
@@ -0,0 +1,62 @@
+package network
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/packetbeat/procs"
+	"github.com/elastic/beats/v7/packetbeat/protos/applayer"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+// PacketData tracks all counters for a given port
+type PacketData struct {
+	Incoming PortsForProtocol
+	Outgoing PortsForProtocol
+}
+
+// ContainsMetrics returns true if the metrics have non-zero data
+func (pd PacketData) ContainsMetrics() bool {
+	return pd.Incoming.TCP > 0 || pd.Incoming.UDP > 0 || pd.Outgoing.TCP > 0 || pd.Outgoing.UDP > 0
+}
+
+// PortsForProtocol tracks counters for TCP/UDP connections
+type PortsForProtocol struct {
+	TCP uint64
+	UDP uint64
+}
+
+// CounterUpdateEvent is sent every time we get new packet data for a PID
+type CounterUpdateEvent struct {
```

Comment:
I think it would be good to review which types in the package need to be exported. I think many of these internal-only structs should not be exported.